prime-router/src/main/kotlin/fhirengine/translation/HL7toFhirTranslator.kt

```diff
@@ -40,16 +79,11 @@ class HL7toFhirTranslator(
     ): HL7MessageModel {
         val messageTemplateType = getMessageTemplateType(hl7Message)
 
-        // if the requested message templates have been previously loaded, return the stored templates.
-        // otherwise, load the templates
-        val messageTemplate = messageTemplates.getOrElse(configFolderPath) {
-            val newMessageTemplate = ResourceReader(ConverterConfiguration(configFolderPath)).messageTemplates
-            messageTemplates[configFolderPath] = newMessageTemplate
-            newMessageTemplate
-        }
+        val messageTemplate = getMessageTemplates()[configFolderPath]
+            ?: throw IllegalArgumentException("Invalid FHIR translator configuration path: $configFolderPath")
```

Comment:
An exception is thrown if an attempt is made to access a configuration path that was not loaded at initialization. Previously, templates were loaded on demand here.